Water Drink Water/src/clients/blazor.wa.tbd/Pages/Register.razor.cs

```diff
@@ -0,0 +1,45 @@
+﻿using blazor.wa.tbd.Services;
+using Microsoft.AspNetCore.Components;
+
+namespace blazor.wa.tbd.Pages;
+
+public partial class Register : ComponentBase
+{
+    private readonly RegistrationFormData formData = new();
+
+    [Inject] public NavigationManager NavigationManager { get; set; } = null!;
+    [Inject] private AuthService AuthService { get; set; } = null!;
+
+    public string ValidationMessage { get; set; } = string.Empty;
+
+    public async Task ExecuteRegistration()
+    {
+        if (formData.IsValid)
+        {
+            var result = await AuthService.Register(formData.Name, formData.Email, formData.Password);
+
+            if (result)
+            {
+                NavigationManager.NavigateTo("/login");
+            }
+
+            ValidationMessage = "Unable to register, please try again.";
+        }
+        else
+        {
+            ValidationMessage = "Please fill out all fields and ensure passwords match.";
```

Comment:
We won't be needing the IF else if we use the Data Annotations.n The Validation Message if the registering itself failed, is great feedback. So I would keep that.